### PR TITLE
F#src5190 add cpp wait for param

### PR DESCRIPTION
**Pull request by giusebar**
## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropiate.

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR. This is a reminder of what we should look for before merging this code. I have:

- [ ] Read and follow the [contributing guidelines](https://github.com/shadow-robot/sr_documentation/blob/master/CONTRIBUTING.md).
- [ ] Checked that all tests pass with my changes
- [ ] Added tests (automatic or manual) that prove the fix is effective or that the feature works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added the corresponding license to each file and add the current year of any one that you modified.
- [ ] Tested on real hardware (if appropriate)

### CODE REVIEW

File: sr_utilities_common/include/sr_utilities_common/wait_for_param.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+* Copyright (C) 2019 Shadow Robot Company Ltd - All Rights Reserved. Proprietary and Confidential.
```
Comment (mikramarc) [inline sr_utilities_common/include/sr_utilities_common/wait_for_param.h]: date, and its public

**Review by ghost — APPROVED** [review]

File: sr_utilities_common/README.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ This script allows to delay the launch of ros launch files and nodes. It checks
 The node **requires** the following parameters:
 - package_name: the package in which the executable is stored
 - executable_name: this is the name of the launch file we want to launch (must end with .launch) or the name of the node we want to run
-- timeout: after how long the script should give up waiting for the required conditions
+- timeout: after how long the script should give up waiting for the required conditions. If not set the script will wait forever for the condition to be met.
```
Comment (mikramarc) [inline sr_utilities_common/README.md]:
```suggestion
- timeout: after how long the script should give up waiting for the required conditions. If not set or set to zero or less, the script will wait indefinitely for the condition to be met.
```

File: sr_utilities_common/scripts/sr_utilities_common/conditional_delayed_rostool.py [hunk 3]
```diff
@@ -69,13 +67,14 @@ def wait_for_conditions(conditions_to_satisfy, timeout):
...
         if all(satisfied for satisfied in all_conditions_satisfied_list):
             all_conditions_satisfied = True
-        if (round(rospy.Time.now().to_sec(), 1) == round(time.to_sec(), 1)):
-            rospy.logerr("Timeout of {}s exceeded".format(timeout))
-            for condition_type, condition in conditions_to_satisfy.iteritems():
-                if condition.missing_elements:
-                    rospy.logerr('Could not find the following {}s: {}'.format(condition_type,
-                                                                               condition.missing_elements))
-            break
+        if timeout > 0:
```
Comment (mikramarc) [inline sr_utilities_common/scripts/sr_utilities_common/conditional_delayed_rostool.py]: if timeout <= 0:
    continue

? would avoid nesting

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: sr_utilities_common/include/sr_utilities_common/wait_for_param.h (hunk 1 above)
Comment (giusebar) [reply sr_utilities_common/include/sr_utilities_common/wait_for_param.h]: wait i had fixed that

Comment (mikramarc) [reply sr_utilities_common/include/sr_utilities_common/wait_for_param.h]: ye, outdated

**Review by mikramarc — APPROVED** [review]

**Pull request merged**